homeassistant/components/switch/zoneminder.py

```diff
@@ -49,26 +47,21 @@ class ZMSwitchMonitors(SwitchDevice):
 
     icon = 'mdi:record-rec'
 
-    def __init__(self, monitor_id, monitor_name, on_state, off_state):
+    def __init__(self, monitor, on_state, off_state):
         """Initialize the switch."""
-        self._monitor_id = monitor_id
-        self._monitor_name = monitor_name
+        self._monitor = monitor
         self._on_state = on_state
         self._off_state = off_state
         self._state = None
 
     @property
     def name(self):
         """Return the name of the switch."""
-        return "%s State" % self._monitor_name
+        return '{}\'s State'.format(self._monitor.name)
```

Comment:
Looks like we have changed the name here by including an apostrophe. That will also change the entity_id and make this a breaking change. Is that necessary?

Reply:
Wait, apostrophes should be removed by the core from the entity_id, so maybe this is ok.

Reply:
The apostrophe addition was to make the friendly name make more sense:
`FrontDoor's State` makes more grammatical sense that `FrontDoors State` (which was the predecessor).

I wanted to drop the s and just make it `FrontDoor State` but didn't because it would change the `entity_id`. I can confirm that the `entity_id` should not change. My switch is still called `switch.frontdoors_state`.

Reply:
Thanks!

Reply:
@mnoorenberghe I'm confused. The `s` was always in the name. The only change I made there was to add an `'`

Can you give an example of what your switch was named before/after this change?

Reply:
That's incorrect, `%s` is for the format identifier for a string so `%s` would be replaced by "Basement" for example. There was no literal 's' before.

Reply:
The previous code was like:
```py
        return "{} State".format(self._monitor_name)
```

Reply:
D'oh you are right. I totally messed that up when changing it to the `.format` approach for string formatting. :(

@MartinHjelmare is this something we want to hotfix? Because it's now a breaking change. But fixing it would make another breaking change.

Reply:
https://github.com/home-assistant/home-assistant/pull/17026 to fix this